src/SymbolDirectories/ManagerTest.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <QCoreApplication>
+#include <filesystem>
+
+#include "SymbolDirectories/Manager.h"
+
+const std::filesystem::path directory0{"/path/to/symbols/directory"};
+const std::filesystem::path directory1{"/home/src/project/build/"};
+const std::filesystem::path directory2{R"(c:\project\build\)"};
+
+constexpr const char* kOrgName = "The Orbit Authors";
+
+namespace orbit_symbol_directories {
+
+TEST(SymbolDirectoriesManager, SetAndGet) {
+  QCoreApplication::setOrganizationDomain(kOrgName);
+  QCoreApplication::setApplicationName("SymbolDirectoryManager.SetAndGet");
+
+  Manager manager{};
+
+  std::vector<std::filesystem::path> directories{
+      directory0,
+      directory1,
+      directory2,
+  };
+
+  manager.SetDirectories(directories);
+  EXPECT_EQ(manager.GetDirectories(), directories);
+}
+
+TEST(SymbolDirectoriesManager, Append) {
+  QCoreApplication::setOrganizationName(kOrgName);
+  QCoreApplication::setApplicationName("SymbolDirectoriesManager.Append");
+
+  Manager manager{};
+
+  std::vector<std::filesystem::path> directories{directory0, directory1};
+  manager.SetDirectories(directories);
+
+  manager.AppendDirectory(directory2);
+  directories.emplace_back(directory2);
+
+  EXPECT_EQ(manager.GetDirectories(), directories);
+}
+
+TEST(SymbolDirectoriesManager, SaveLoadAndClear) {
+  QCoreApplication::setOrganizationName(kOrgName);
+  QCoreApplication::setApplicationName("SymbolDirectoriesManager.SaveAndLoad");
+
+  std::vector<std::filesystem::path> directories{directory0, directory1, directory2};
+  {
+    Manager manager{};
+    manager.SetDirectories(directories);
+  }
+
+  {
+    Manager manager{};
+    EXPECT_EQ(manager.GetDirectories(), directories);
+    manager.SetDirectories({});
+  }
+
+  {
+    Manager manager{};
+    EXPECT_TRUE(manager.GetDirectories().empty());
+  }
```

Comment:
what if 2 managers exists at the same time and you do set in one, and get in another?

Reply:
In the way it is currently implemented, it depends on the exact order the construction, gets and sets are happening
```
Manager manager0{};
manager0.SetDirs(dir0);
Manager manager1{};
manager1.GetDirs(); // This will return vector with dir0
```
```
Manager manager0{};
Manager manager1{};
manager0.SetDirs(dir0);
manager1.GetDirs(); // This will return empty vector
```

I could change the behaviour and call `LoadDirectories` on every call of `GetDirectories`. This would mean I would drop `directories_` all together and Manager would become an even more shallow wrapper around QSettings. I dont have a strong preference aka light preference to leave it as is. wdyt?